site/source/docs/api_reference/wasm_audio_worklets.rst

```diff
@@ -0,0 +1,181 @@
+
+    // Connect it to audio context destination
+    EM_ASM({emscriptenGetAudioObject($0).connect(emscriptenGetAudioObject($1).destination)},
+      wasmAudioWorklet, audioContext);
+
+    // Resume context on mouse click
+    emscripten_set_click_callback("canvas", (void*)audioContext, 0, OnCanvasClick);
+  }
+
+4. The code to resume the audio context on click looks like this:
+
+.. code-block:: cpp
+
+  EM_BOOL OnCanvasClick(int eventType, const EmscriptenMouseEvent *mouseEvent, void *userData)
+  {
+    EMSCRIPTEN_WEBAUDIO_T audioContext = (EMSCRIPTEN_WEBAUDIO_T)userData;
+    if (emscripten_audio_context_state(audioContext) != AUDIO_CONTEXT_STATE_RUNNING) {
+      emscripten_resume_audio_context_sync(audioContext);
+    }
+    return EM_FALSE;
+  }
+
+5. Finally we can implement the audio callback that is to generate the noise:
+
+.. code-block:: cpp
+
+  #include <emscripten/em_math.h>
+
+  EM_BOOL GenerateNoise(int numInputs, const AudioSampleFrame *inputs,
+                        int numOutputs, AudioSampleFrame *outputs,
+                        int numParams, const AudioParamFrame *params,
+                        void *userData)
+  {
+    for(int i = 0; i < numOutputs; ++i)
+      for(int j = 0; j < 128*outputs[i].numberOfChannels; ++j)
+        outputs[i].data[j] = emscripten_random() * 0.2 - 0.1; // Warning: scale down audio volume by factor of 0.2, raw noise can be really loud otherwise
+
+    return EM_TRUE; // Keep the graph output going
+  }
+
+And that's it! Compile the code with the linker flags ``-sAUDIO_WORKLET=1 -sWASM_WORKERS=1`` to enable targeting AudioWorklets.
+
+Synchronizing audio thread with the main thread
+===============================================
+
+Wasm Audio Worklets API builds on top of the Emscripten Wasm Workers feature. This means that the Wasm Audio Worklet thread is modeled as if it was a Wasm Worker thread.
+
+To synchronize information between an Audio Worklet Node and other threads in the application, there are two options:
+
+1. Leverage the Web Audio "AudioParams" model. Each Audio Worklet Processor type is instantiated with a custom defined set of audio parameters that can affect the audio computation at sample precise accuracy. These parameters are passed in the ``params`` array into the audio processing function.
+
+The main browser thread that created the Web Audio context can adjust the values of these parameters whenever desired. See `MDN function: setValueAtTime <https://developer.mozilla.org/en-US/docs/Web/API/AudioParam/setValueAtTime>`_ .
+
+2. Data can be shared with the Audio Worklet thread using GCC/Clang lock-free atomics operations, Emscripten atomics operations and the Wasm Worker API thread synchronization primitives. See :ref:`wasm_workers` for more information.
+
+More Examples
+=============
+
+See the directory tests/webaudio/ for more code examples on Web Audio API and Wasm AudioWorklets.
```

Comment:
Please add a note in this file that you can use this with pthreads - that is, all of the above works with  `-pthread -s WASM_WORKERS` during link. I think that might avoid some possible confusion here.

Reply:
Check, added a paragraph about that.